californium-core/src/main/java/org/eclipse/californium/core/network/stack/BlockwiseLayer.java

```diff
@@ -1041,27 +1041,32 @@ private Block2BlockwiseStatus getBlock2Status(final KeyUri key) {
 	}
 
 	private Block1BlockwiseStatus clearBlock1Status(final KeyUri key) {
+		int size;
+		Block1BlockwiseStatus removedTracker;
 		synchronized (block1Transfers) {
-			Block1BlockwiseStatus removedTracker = block1Transfers.remove(key);
-			if (removedTracker != null) {
-				LOGGER.debug("removing block1 tracker [{}], block1 transfers still in progress: {}",
-						new Object[] { key, block1Transfers.size() });
-				removedTracker.setComplete(true);
-			}
-			return removedTracker;
+			removedTracker = block1Transfers.remove(key);
+			size = block1Transfers.size();
```

Comment:
I hesitated to enclose this line with a if LOGGER.isDebugEnabled() but the size() is just an access to an attribute so I'm no sure if this is worth.

Reply:
Don't make it too complex. So leave it, as it is here.